libraries/AP_Airspeed/AP_Airspeed_ND210.cpp

```diff
@@ -0,0 +1,302 @@
+ * The value of the Rate Control Register is
+ * the divisor of the 444 Hz internal data rate. Since a divisor of
+ * zero is not possible, a zero value will select the auto-select
+ * rate mode. In this mode, the rate is selected based on the
+ * selected bandwidth limit. The auto rate value is roughly 3x
+ * the corner frequency of the selected bandwidth limit in all
+ * auto selected rates (where possible).
+ * 00000000 = AutoSelect (based on BW)
+ * 00000001 = 444Hz
+ * 00000010 = 222Hz ...
+ */
+static constexpr uint8_t RATE_CONTROL_REGISTER = 0x00; // Auto Select aka (BW = 1Hz, so Rate will be 3Hz)
+
+extern const AP_HAL::HAL &hal;
+
+AP_Airspeed_ND210::AP_Airspeed_ND210(AP_Airspeed &_frontend, uint8_t _instance) :
+    AP_Airspeed_Backend(_frontend, _instance)
+{
+}
+
+/*
+  send a 16 bit command code
+ */
+bool AP_Airspeed_ND210::_send_configuration_command()
+{
+    uint8_t b[2] {MODE_CONTROL_REGISTER, RATE_CONTROL_REGISTER};
+    return _dev->transfer(b, 2, nullptr, 0);
+}
+
+// probe and initialise the sensor
+bool AP_Airspeed_ND210::init()
+{
+    _dev = hal.i2c_mgr->get_device(get_bus(), ND210_I2C_ADDR);
+    if (!_dev) {
+        GCS_SEND_TEXT(MAV_SEVERITY_ERROR, "ND210 no sensor found at 0x%02X", ND210_I2C_ADDR);
+        return false;
+    }
+    _dev->get_semaphore()->take_blocking();
+    // lots of retries during probe
+    _dev->set_retries(10);
+
+    // set Mode and Rate modes
+    if (!_send_configuration_command()) {
+        _dev->get_semaphore()->give();
+        GCS_SEND_TEXT(MAV_SEVERITY_ERROR, "ND210[%u]: Init Send Command Failed.", get_instance());
+        return false;
+    }
+
+    _dev->get_semaphore()->give();
+
+    switch(MODE_CONTROL_REGISTER & 0x07) {
+        case 0b000:
+        case 0b001:
+            _selected_pressure_range = 0.25;
+            break;
+        case 0b010:
+            _selected_pressure_range = 0.5;
+            break;
+        case 0b011:
+            _selected_pressure_range = 1.0;
+            break;
+        case 0b100:
+            _selected_pressure_range = 2.0;
+            break;
+        case 0b101:
+            _selected_pressure_range = 4.0;
+            break;
+        case 0b110:
+            _selected_pressure_range = 5.0;
+            break;
+        case 0b111:
+            _selected_pressure_range = 10.0;
+            break;
+    }
+
+    /*
+      this sensor uses zero offset and skips cal
+     */
+#if 0  // Airspeed seems more accurate without these.
+    set_use_zero_offset();
+    set_skip_cal();
+    set_offset(0);
+#endif
+
+    _dev->set_device_type(uint8_t(DevType::ND210));
+    set_bus_id(_dev->get_bus_id());
+
+    // drop to 2 retries for runtime
+    _dev->set_retries(2);
+
+    _dev->register_periodic_callback(20000, FUNCTOR_BIND_MEMBER(&AP_Airspeed_ND210::_timer, void));
+
+    _print_info();
+
+    return true;
+}
+
+void AP_Airspeed_ND210::_print_info()
+{
+#if HAL_GCS_ENABLED
```

Comment:
```suggestion
#if HAL_GCS_ENABLED
void AP_Airspeed_ND210::_print_info()
{
```

Avoid having the method at all if it's not used

Reply:
done